pkg/kubelet/server/server.go

```diff
@@ -716,6 +716,13 @@ func (s *Server) getContainerLogs(request *restful.Request, response *restful.Re
 		response.WriteError(http.StatusBadRequest, fmt.Errorf(`{"message": "Unable to decode query."}`))
 		return
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodLogsQuerySplitStreams) {
+		// Even with defaulters, logOptions.Stream can be nil if no arguments are provided at all.
+		if logOptions.Stream == nil {
+			response.WriteError(http.StatusBadRequest, fmt.Errorf(`{"message": "Missing stream type"}`))
```

Comment:
I'm dopn't think you can hard-fail here.  If you have some back-rev component that is using this API directly, it might not know about this new query-param, and so things that used to work will break, through no fault of their own.  Bad.

I don't know if any such components exist, but this is tantamount to adding a new required field, so I think all you can do is to "bake in" the default value.  If ever there is a v2 API, you could theoretically have a different default.